3rdparty/libprocess/src/posix/subprocess.hpp

```diff
@@ -131,7 +131,12 @@ static void handleWhitelistFds(const std::vector<int_fd>& whitelist_fds)
           }
         }
 
-        if (!found) {
+        Try<bool> isCloexec = os::isCloexec(fd);
```

Comment:
os::isCloexec is not async signal safe in the error path as os::strerror allocate memory and may require locale.

I notice the same issue above. We should follow up and clean this up.